blocks/firefly-gallery/firefly-gallery.js

```diff
@@ -203,3 +227,18 @@ export default async function decorate(block) {
     }, 2000);
   }
 }
+
+export default async function decorate(block) {
+  loadIms()
+    .then(async () => {
```

Comment:
I wouldn't wait for ims to load to show the whole gallery. the ims piece is only needed for favorites if someone clicks on the favorite icon. so that's when i would check for the window.adobeIMS object and send them to login if they are not logged in already.